src/admin/components/views/collections/Edit/Default.tsx

```diff
@@ -78,14 +81,29 @@ const DefaultEditView: React.FC<Props> = (props) => {
     isEditing && `${baseClass}--is-editing`,
   ].filter(Boolean).join(' ');
 
-  const onSave = useCallback((json) => {
+  const onSave = useCallback(async (json) => {
     if (typeof onSaveFromProps === 'function') {
+      if (auth && id === user.id) {
+        const request = await requests.post(`${serverURL}${api}/${collection.slug}/refresh-token`, {
+          headers: {
+            'Accept-Language': i18n.language,
+          },
+        });
+
+        if (request.status === 200) {
```

Comment:
I would suggest to handle /me and /refresh-token in one method. There are currently 3 slightly different handlers and all without error handling (offline / gateway timeout)

* components/views/collections/Edit/Default.tsx:93
* components/utilities/Auth/index.tsx:59
* components/utilities/Auth/index.tsx:106

Reply:
👍 I went ahead and added some error handling to the network requests in the auth provider